x-pack/filebeat/module/fortinet/firewall/ingest/traffic.json

```diff
@@ -0,0 +1,506 @@
+{
+  "description": "Pipeline for parsing fortinet firewall logs (traffic pipeline)",
+  "processors": [
+    {
+      "set": {
+        "field": "event.kind",
+        "value": "event"
+      }
+    },
+    {
+      "set": {
+        "field": "event.action",
+        "value": "{{fortinet.firewall.action}}",
+        "if": "ctx.fortinet?.firewall?.action != null"
+      }
+    },
+    {
+      "set": {
+        "field": "event.outcome",
+        "value": "success",
+        "if": "ctx.fortinet?.firewall?.action != null"
+      }
+    },
+    {
+      "append": {
+        "field": "event.category",
+        "value": "network"
+      }
+    },
+    {
+      "append": {
+        "field": "event.type",
+        "value": "connection"
+      }
+    },
+    {
+      "append": {
+        "field": "event.type",
+        "value": "start",
+        "if": "ctx.fortinet?.firewall?.action == 'start'"
+      }
+    },
+    {
+      "append": {
+        "field": "event.type",
+        "value": "end",
+        "if": "ctx.fortinet?.firewall?.action != 'start' && ctx.fortinet?.firewall?.action != null"
+      }
+    },
+    {
+      "append": {
+        "field": "event.type",
+        "value": "error",
+        "if": "['dns', 'ip-conn'].contains(ctx.fortinet?.firewall?.action)"
```

Comment:
this is interesting, it might be worth switching format to yaml so we can easily have a comment that says why dns & ip-conn actions are errors.

Reply:
Next commit will come in YAML, added a JSON commit first so its easier to see the changes I did between the newest JSON and the earlier. The mapping however seems to be correct.

Reply:
This seems to have been a typo from me.